tracing/src/test/java/com/palantir/remoting3/tracing/TracersTest.java

```diff
@@ -244,6 +244,53 @@ public void testWrapRunnableWithNewTrace_traceStateRestoredWhenThrows() throws E
         assertThat(Tracer.getTraceId()).isEqualTo(traceIdBeforeConstruction);
     }
 
+    @Test
+    public void testWrapRunnableWithTrace_traceStateInsideRunnableUsesGivenTraceId() {
+        String traceIdBeforeConstruction = Tracer.getTraceId();
+
+        List<String> traceIds = Lists.newArrayList();
+
+        String traceIdToUse = "someTraceId";
+        Runnable wrappedRunnable = Tracers.wrapWithTrace(() -> {
+            traceIds.add(Tracer.getTraceId());
+        }, traceIdToUse);
+
+        wrappedRunnable.run();
+        wrappedRunnable.run();
+
+        String traceIdFirstCall = traceIds.get(0);
+        String traceIdSecondCall = traceIds.get(1);
+
+        String traceIdAfterCalls = Tracer.getTraceId();
```

Comment:
inline

Reply:
It's used twice. Also, the assertions are more readable this way. Renamed to `traceIdAfterCall` since I'm doing just one call now.